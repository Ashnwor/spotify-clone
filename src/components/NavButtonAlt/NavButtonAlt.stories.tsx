import NavButtonAlt from './NavButtonAlt';

export default {
  title: 'Topbar Navigation Button',
};

export const topbarNavigationButton: React.FC = () => {
  return (
    <div style={{ height: 300, position: 'relative' }}>
      <NavButtonAlt direction="left" />
      <NavButtonAlt direction="right" />
    </div>
  );
};
