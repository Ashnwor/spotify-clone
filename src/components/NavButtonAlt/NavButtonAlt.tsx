import styles from './NavButtonAlt.module.scss';
import { Back, Forward } from '../icons';

interface Props {
  direction: string;
}

const NavButtonAlt: React.FC<Props> = ({ direction }) => {
  return (
    <div className={styles['nav-button']}>
      {direction === 'left' && <Back className={styles['nav-icon']} />}
      {direction === 'right' && <Forward className={styles['nav-icon']} />}
    </div>
  );
};

export default NavButtonAlt;
