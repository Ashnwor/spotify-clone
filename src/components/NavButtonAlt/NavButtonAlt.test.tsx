import { shallow } from 'enzyme';
import NavButtonAlt from './NavButtonAlt';

it('runs correctly for left', () => {
  const wrapper = shallow(<NavButtonAlt direction="left" />);
  expect(wrapper).toMatchSnapshot();
});

it('runs correctly for right', () => {
  const wrapper = shallow(<NavButtonAlt direction="right" />);
  expect(wrapper).toMatchSnapshot();
});
