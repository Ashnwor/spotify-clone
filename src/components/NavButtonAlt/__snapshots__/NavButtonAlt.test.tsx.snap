// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs correctly for left 1`] = `
<div
  className="nav-button"
>
  <SvgBack
    className="nav-icon"
  />
</div>
`;

exports[`runs correctly for right 1`] = `
<div
  className="nav-button"
>
  <SvgForward
    className="nav-icon"
  />
</div>
`;
